### PR TITLE
Release longhorn v1.3.2 into Rancher 2.6

### PR DESCRIPTION
**Pull request by PhanLe1010**
https://github.com/rancher/rancher/issues/38667

### CODE REVIEW

**Review by derekbit — APPROVED** [review]

**Comment by innobead** [comment]
@PhanLe1010 suggests using the same PR for each milestone release testing, so you don't need to create a different one for each.

**Comment by PhanLe1010** [comment]
> @PhanLe1010 suggests using the same PR for each milestone release testing, so you don't need to create a different one for each.

Yeah, I am using the same PR (just changed the title)

**Comment by prachidamle** [comment]
@PennyScissors This PR is for the OOB release to 2.6 branch after v2.6.9 is out.

**Comment by khushboo-rancher** [comment]
Validation - PASS with longhornio image

Validated installation, upgrade, volume attach detach, uninstallation.

File: release.yaml
Comment (shuo-wu) [inline release.yaml:26]: Not sure why these are removed (by the script I think)...

Comment (PhanLe1010) [reply release.yaml:26]: It need to be removed since we are only updating Longhorn in this PR.

**Comment by PennyScissors** [comment]
@PhanLe1010 This chart should follow the normal process and go to dev-v2.6 first. Then a separate PR to move the assets from dev-v2.6 to release-v2.6 for the OOB release.

**Comment by PhanLe1010** [comment]
@PennyScissors  The information here is new to the Longhorn team. We have been doing PR into release-2.6 so far and there wasn't any warning. For this PR, it has been created quite a long time but no warning message and QA already spent the effort to verify it. Can we move forward with this OOB?

In the future,

> @PhanLe1010 This chart should follow the normal process and go to dev-v2.6 first. Then a separate PR to move the assets from dev-v2.6 to release-v2.6 for the OOB release (basically move the contents of package to the release branch, add it to release.YAML, and run make charts).

Can you point me to the official document detailing the steps here? I am not clear about `basically move the contents of package to the release branch, add it to release.YAML, and run make charts` ?

**Comment by PennyScissors** [comment]
@PhanLe1010 We don't have an official document, but we are currently working on it. You can keep the PR as is, but the chart needs to be brought back into the dev branch after the OOB release so fix the validation workflow in the dev branch.

**Comment by PhanLe1010** [comment]
> Usually charts go to dev-v2.6 first, get tested by QA, and then go to release-v2.6 for the OOB release.

Can I have question about how to move chart from `dev-v2.6` to `release-v2.6`? Do we:
1. Copy the `./asset/chart-name` and `./charts/chart-name` from `dev-v2.6` to `release-v2.6`. And run `make charts` ?
2. Or we move `./package/chart-name` to from `dev-v2.6` to `release-v2.6`. And run `make charts`?

**Comment by PennyScissors** [comment]
@PhanLe1010 What I explained before about moving the contents from the package directory and running make charts is for the case where a chart is merged into dev-v2.6 first because we usually need the chart in dev-v2.6 for QA to move forward with validation, and then is a matter of moving those to the release branch. However, seems like Longhorn is a special case because it gets validated before it makes it into the rancher/charts repository. 

Merging this PR will release the chart OOB as it is made to the release branch. I think all you need to do to bring it to dev is a `git merge release-v2.6` into your dev-v2.6 branch and fix any conflicts that arise (usually release.yaml cause conflicts).

**Comment by PhanLe1010** [comment]
> What I explained before about moving the contents from the package directory and running make charts is for the case where a chart is merged into dev-v2.6 first because we usually need the chart in dev-v2.6 for QA to move forward with validation, and then is a matter of moving those to the release branch.

Yeah, I want to ask more about this process so Longhorn can follow it in the future. How do we copy charts from `dev-v2.6` to `release-v2.6`? Do we
1. Copy the `./asset/chart-name` and `./charts/chart-name` from `dev-v2.6` to `release-v2.6`. And run `make charts`? OR
2. Or we move `./package/chart-name` to from `dev-v2.6` to `release-v2.6`. And run `make charts`?

> Merging this PR will release the chart OOB as it is made to the release branch. I think all you need to do to bring it to dev is a git merge release-v2.6 into your dev-v2.6 branch and fix any conflicts that arise (usually release.yaml cause conflicts).

Got it. Thanks

**Comment by innobead** [comment]
@PennyScissors as @PhanLe1010 said, this new flow is different from the last time we have been told, so it's better to clarify the detailed steps, so the team can follow them.

However, a general question, why not just merge to the dev-2.6 and then cherry-pick later to release-2.6 by the charts repo release captain, so for the contributor(@PhanLe1010 in this case), we should just focus on this PR only w/ requiring the knowledge about the following release process, because the charts team will move that forward. WDYT? @PennyScissors @prachidamle

**Comment by PennyScissors** [comment]
@innobead @PhanLe1010 We are short on time so please disregard my previous comments. Lets just continue with the same process you are familiar with and I'll take care of what comes after merging this PR. We are currently working on the documentation for OOB releases, so once it is ready I'll make sure to share it with you and answer any further questions you may have.

**Comment by PhanLe1010** [comment]
> But just one part is different from my understanding, why longhorn-crd/longhorn-1.3/generated-changes/exclude only include few files? I thought all files should be included execpte crd template?

All files are excluded as you expected https://github.com/rancher/charts/tree/e9768dab5033d9cc0257243fd5627c19b7ffc69b/packages/longhorn-crd/longhorn-1.3/generated-changes/exclude. GitHub PR only displays the files that have changes (not all files in ./excluded folder)

**Comment by PennyScissors** [comment]
Can you split this PR into 2 commits to make it easier to review? First, the changes made in packages/ and release.yaml, and second the output of make charts.

**Review by innobead — APPROVED** [review]

**Comment by PhanLe1010** [comment]
> Can you split this PR into 2 commits to make it easier to review? First, the changes made in packages/ and release.yaml, and second the output of make charts.

Done

**Comment by khushboo-rancher** [comment]
I see the [mirrored image PR](https://github.com/rancher/image-mirror/pull/294) is merged. I'll do a final check once the @PennyScissors approve this.

**Review by PennyScissors — APPROVED** [review]

**Comment by khushboo-rancher** [comment]
Validated the PR with Rancher v2.6.9.
Validation : PASS

Validated installation, upgrade, uninstallation and some workload deployment. Found an [issue](https://github.com/rancher/dashboard/issues/7253) in Rancher.

**Comment by MKlimuszka** [comment]
> Validated the PR with Rancher v2.6.9. Validation : PASS
> 
> Validated installation, upgrade, uninstallation and some workload deployment. Found an [issue](https://github.com/rancher/dashboard/issues/7253) in Rancher.

Does that issue block release? Or are we approved for release?

**Comment by PhanLe1010** [comment]
No, the above issue does not block this release.

**Pull request merged**
